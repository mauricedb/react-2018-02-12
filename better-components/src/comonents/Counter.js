import React, { Component } from 'react';

class Counter extends Component {
  state = { counter: 0 };

  onClick = () => {
    this.setState({
      counter: this.state.counter + 1
    });
  };

  render() {
    const { counter } = this.state;
    return (
      <div>
        <p>The count = {counter}</p>
        <button onClick={this.onClick}>Increment</button>
      </div>
    );
  }
}

export default Counter;
