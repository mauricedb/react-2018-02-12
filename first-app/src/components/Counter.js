import React, { Component } from "react";

class CounterUI extends Component {
    element = null;

  onClick = (e) => {
      console.log(e)
    this.props.increment(5);
  };
  componentDidMount(){
      console.log('The element is:', this.element)
      this.element.style.color = 'red'
  }
  render() {
    console.log("Counter  rendering");
    const { counter } = this.props;
    return (
      <div ref={e => this.element= e}>
        The count = {counter}
        <button onClick={this.onClick}>Increment</button>
      </div>
    );
  }
}

class Counter extends Component {
  state = { counter: 0, date: new Date() };

  increment = (value = 1) => {
    const { counter } = this.state;

    this.setState({
      counter: counter + value,
      date: new Date(),
      x: 12
    });
  };

  render() {
    return (
      <CounterUI counter={this.state.counter} increment={this.increment} />
    );
  }
}

export default Counter;
