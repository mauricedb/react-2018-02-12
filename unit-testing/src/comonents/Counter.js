import React, { Component } from 'react';

class Counter extends Component {
  state = { counter: 0 };

  onClick = () => {
    this.setState(state => ({
      counter: state.counter + 1
    }), () => {
      console.log(this.state)
    });
    
    this.setState(state => ({
      counter: state.counter + 1
    }), () => {
      console.log(this.state)
    });
  };

  render() {
    const { counter } = this.state;
    return (
      <div>
        <p>The count = {counter}</p>
        <button onClick={this.onClick}>Increment</button>
      </div>
    );
  }
}

export default Counter;
