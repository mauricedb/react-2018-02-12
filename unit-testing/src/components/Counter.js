import React, { Component } from 'react';

class Counter extends Component {
  state = { counter: 0 };

  onClick = () => {
    this.setState(state => ({
      counter: state.counter + 1
    }));
  };

  render() {
    const { counter } = this.state;
    return (
      <div>
        <p>The count = {counter}</p>
        <button onClick={this.onClick}>Increment</button>
      </div>
    );
  }
}

export default Counter;
