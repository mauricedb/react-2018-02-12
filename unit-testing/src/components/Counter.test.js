import React from 'react';
import { shallow } from 'enzyme';
import reactTestRenderer from "react-test-renderer";

import Counter from './Counter';

describe('The Counter component', () => {
  let greeter;

  beforeEach(() => {
    greeter = shallow(<Counter />);
  });

  it('should render the initial count', () => {
    expect(greeter.find('p').text()).toBe('The count = 0');
  });

  it('should increment the count in click', () => {
    greeter.find('button').simulate('click');
    expect(greeter.find('p').text()).toBe('The count = 1');
  });

  it('should increment the count in the state click', () => {
    greeter.find('button').simulate('click');
    expect(greeter.state('counter')).toBe(1);
  });

  it('should render 5 aftert setting state', () => {
    greeter.setState({ counter: 5 });
    expect(greeter.find('p').text()).toBe('The count = 5');
  });

  it("renders the same tree", () => {
    const tree = reactTestRenderer
      .create(<Counter />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

});
