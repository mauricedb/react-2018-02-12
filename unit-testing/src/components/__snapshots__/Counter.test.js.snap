// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Counter component renders the same tree 1`] = `
<div>
  <p>
    The count = 
    0
  </p>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
</div>
`;
